client/src/pages/private/PSIViewDetails.js

```diff
@@ -54,74 +75,76 @@ export default ({ match }) => {
-      fetchCohorts(psiID);
-    } else {
+      openToast({
+        status: ToastStatus.Success,
+        message: `Cohort '${cohort.cohortName}' added successfully`,
+      });
+      const cohorts = await fetchCohorts({ psiId: psiID });
+      setCohorts(cohorts);
+    } catch (error) {
       openToast({
         status: ToastStatus.Error,
-        message: response.error || response.statusText || 'Server error',
+        message: error.message || 'Unable to add cohort',
       });
     }
   };
 
-  const fetchPSI = async (psiID) => {
-    const response = await fetch(`${API_URL}/api/v1/psi/${psiID}`, {
-      headers: {
-        Authorization: `Bearer ${store.get('TOKEN')}`,
-      },
-      method: 'GET',
-    });
+  const handleCohortEdit = async (newCohort) => {
+    try {
+      await editCohort({ cohort: newCohort, cohortId: cohort.id });
 
-    if (response.ok) {
-      const psi = await response.json();
-      setPSI({
-        id: psi.id,
-        instituteName: psi.institute_name,
-        healthAuthority: psi.health_authority,
-        streetAddress: psi.street_address,
-        postalCode: psi.postal_code,
-        city: psi.city,
+      openToast({
+        status: ToastStatus.Success,
+        message: `Cohort '${newCohort.cohortName}' (#${cohort.id}) updated successfully`,
+      });
+      setActiveModalForm(null);
+      setCohort(null);
+      const cohorts = await fetchCohorts({ psiId: psiID });
+      setCohorts(cohorts);
+    } catch (error) {
+      openToast({
+        status: ToastStatus.Error,
+        message: error.message || 'Unable to add cohort',
       });
     }
   };
 
-  const fetchCohorts = async (psiID) => {
-    const response = await fetch(`${API_URL}/api/v1/psi/${psiID}/cohorts/`, {
-      headers: {
-        Authorization: `Bearer ${store.get('TOKEN')}`,
-      },
-      method: 'GET',
-    });
-
-    if (response.ok) {
-      const cohortList = await response.json();
-      setCohorts(cohortList);
-    }
+  const displayEditCohortModal = async (cohort) => {
+    setActiveModalForm('show-cohort');
+    setCohort(cohort);
   };
 
+  // Lifecycle Hooks
   useEffect(() => {
-    fetchPSI(psiID);
-    fetchCohorts(psiID);
-  }, [psiID]);
+    fetchPSI({ psiId: psiID })
+      .then((psiData) => {
+        setPSI(psiData);
+        fetchCohorts({ psiId: psiID })
+          .then((cohortsData) => {
+            setCohorts(cohortsData);
+          })
+          .catch((error) => {
+            setError(error);
+          });
+      })
+      .catch((error) => {
+        setError(error);
+      });
+  }, [psiID, setPSI, setCohorts, setError]);
 
   scrollUp();
   return (
-    <>
+    <Page isAutoHeight={true}>
       <Dialog
         title={
-          activeModalForm === 'edit-psi' ? `Edit PSI (${psi.instituteName})` : `Create New Cohort`
+          activeModalForm === 'edit-psi'
+            ? `Edit PSI (${psi.instituteName})`
+            : cohort
+            ? `Edit Cohort (${cohort.cohort_name})`
```

Comment:
minor nitpick not worth worrying about, but in future I think it would be helpful if we indent nested cases further for clarity :)